### PR TITLE
PKI: Fix small routing issues with SecretListHeader

### PR DESCRIPTION
**Pull request by Monkeychip**
Adopting the routing/display pattern used in roles for keys, certificates and issuers.

Pattern: Adding the parent model to the roles.index (or issuers.index, etc) route so I can use `SecretListHeader` in that hbs file instead of having to use it higher up in the roles.hbs file. 

I'm keeping the index path in the `routes.js` file even though it's not required. For me, I prefer this because it indicates to me that I'm using the index file. But if folks in the future prefer to remove it, you do you. 

```
this.route('roles', function () { // no hbs file anymore (previously displayed just the SecretListHeader
    this.route('index', { path: '/' }); // main hbs file that shows the list view, header and toolbars.
    this.route('create');
    this.route('role', { path: '/:id' }, function () {
      this.route('details');
      this.route('edit');
    });
  });
```

### CODE REVIEW

File: ui/lib/pki/addon/controllers/issuers/index.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { next } from '@ember/runloop';
+import { getOwner } from '@ember/application';
 
 export default class PkiRolesIssuerController extends Controller {
+  get mountPoint() {
```
Comment (Monkeychip) [inline ui/lib/pki/addon/controllers/issuers/index.js:7]: in ember engines, we use mountPoints in the LinkedBlock component. It's used for the @linkPrefix param.

File: ui/lib/pki/addon/routes/issuers/index.js [hunk 2]
```diff
@@ -6,15 +6,23 @@ export default class PkiIssuersIndexRoute extends Route {
...
 
     return this.store
       .query('pki/pki-issuer-engine', { backend: this.secretMountPath.currentPath })
+      .then((issuersModel) => {
+        return { issuersModel, parentModel: this.modelFor('issuers') };
+      })
       .catch((err) => {
         if (err.httpStatus === 404) {
-          return [];
+          return { parentModel: this.modelFor('issuers') };
```
Comment (Monkeychip) [inline ui/lib/pki/addon/routes/issuers/index.js:25]: needed for the EmptyState. This model returns information not on this model but for the parent model that is needed for links if any exist in the emptyState. (The empty states are conditional. The language and links change if the pki engine has been mounted and that part of the work has not been done yet).

File: ui/lib/pki/app/components/pki-key-usage.js [hunk 3]
```diff
@@ -1 +0,0 @@
-export { default } from 'pki/components/pki-key-usage';
```
Comment (Monkeychip) [inline ui/lib/pki/app/components/pki-key-usage.js:1]: I used the cli command `ember g component pki-key-usage -in-repo pki`. this command always adds an additional app folder that we don't use in our engine setup. I missed committing this directory when I created this component.

File: ui/lib/pki/addon/routes/certificates/index.js [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
   model() {
     return this.store
       .query('pki/pki-certificate-engine', { backend: this.secretMountPath.currentPath })
+      .then((certificateModel) => {
+        return { certificateModel, parentModel: this.modelFor('certificates') };
+      })
       .catch((err) => {
         if (err.httpStatus === 404) {
-          return [];
+          return { parentModel: this.modelFor('certificates') };
```
Comment (Monkeychip) [inline ui/lib/pki/addon/routes/certificates/index.js:22]: needed for the EmptyState. The issuers.index or roles.index, etc routes return a model that does not have the information needed for certain links. That's why we have to fetch the parentModel (a secret-engine model) which returns information we need in the EmptyState and SecretListHeader, which will show even if we get a 404 on the LIST request. (The empty states are conditional. The language and links change if the pki engine has been mounted and that part of the work has not been done yet).

File: ui/lib/pki/addon/templates/certificates/index.hbs [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+    text=this.model.parentModel.id
+    path="vault.cluster.secrets.backend.list-root"
+    model=this.model.parentModel.id
+  }}
+  @isEngine={{true}}
+/>
+{{outlet}}
 <Toolbar>
-  {{#if (gt this.model.length 0)}}
+  {{#if (gt this.model.certificateModel.length 0)}}
```
Comment (zofskeez) [inline ui/lib/pki/addon/templates/certificates/index.hbs:13]: If this is always an array you could just look for length as truthy:
`{{if this.model.certificateModel.length}}`

There are some instances I see below using `gt` if you decide to change it.

**Review by zofskeez — APPROVED** [review]
LGTM thanks! I left a comment about using the `gt` helper but just a suggestion.

File: ui/lib/pki/addon/templates/certificates/index.hbs (hunk 5 above)
Comment (Monkeychip) [reply ui/lib/pki/addon/templates/certificates/index.hbs:13]: Good idea! Thanks.

**Pull request merged**